server/game/cards/characters/01/melisandre.js

```diff
@@ -4,7 +4,8 @@ class Melisandre extends DrawCard {
     setupCardAbilities(ability) {
         this.reaction({
             when: {
-                onCardEntersPlay: (e, card) => card.controller === this.controller && card.hasTrait('R\'hllor')
+                onCardPlayed: (event, player, card) =>
```

Comment:
Oops - Mel is only supposed to work on marshaling anyway (which the implementation doesn't currently enforce). My criticism of the onCardPlayed event still stands, though - it doesn't match the language of the rules.